@xen-orchestra/xapi/index.mjs

```diff
@@ -88,6 +88,20 @@ function onRetry(error) {
     })
   } catch (error) {}
 }
+function disconnectBeforeRetry(error) {
+  logErrorBeforeRetry.apply(this, arguments)
+  const { arguments: args, this: self } = this
+  const vdiRef = args[0]
+  return self.VDI_disconnectFromControlDomain(vdiRef).catch(disconnectError =>
+    warn('VDI_disconnectFromControlDomain failed', {
+      attemptNumber: this.attemptNumber,
+      delay: this.delay,
+      error,
+      disconnectError,
+      fn: this.fn.name,
```

Comment:
I think you can just log `disconnectError` here, everything else is already in the logs, and duplicating it won't make it easier to read.

Also, use the `error` name for the property.